source/gui/addonStoreGui/viewModels/store.py

```diff
@@ -604,6 +611,37 @@ def _getAvailableAddonsInBG(self):
 		core.callLater(delay=0, callable=self.detailsVM.updated.notify, addonDetailsVM=self.detailsVM)
 		log.debug("completed refresh")
 
+	def cancelInstallAddon(self, listItemVM: AddonListItemVM[_AddonStoreModel]):

```

Comment:
I think this function would be clearer if it was split up into if/else statements to handle the pending download and pending install case. i.e. `if listItemVM not in addonDataManager._downloadsPendingCompletion`